test/gzip.js

```diff
@@ -56,6 +59,11 @@ test('handles gzip error', async t => {
 	t.is(err.name, 'ReadError');
 });
 
+test('decompress option opts out of decompressing', async t => {
+	const response = await got(s.url, {decompress: false, encoding: null});
```

Comment:
Fixed in https://github.com/sindresorhus/got/pull/320/commits/c99b2500f46673fd4ffa13d8e0e698d6d4b74e50